.github/workflows/e2e-nightly.yaml

```diff
@@ -0,0 +1,156 @@
+              .spec.image = env(CATALOG_IMAGE) |
+              .metadata.name = env(OPERATOR_PKG_NAME)
+              ' ${CATALOG_PATH}
+
+      - name: Create catalog source
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: create -n openshift-marketplace -f ${{ env.CATALOG_PATH }}
+
+      - name: Setup operator group yaml
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq e -i '.spec.targetNamespaces[0] = env(NAMESPACE)' ${OG_PATH}
+
+      - name: Create operator group
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: create -n ${{ env.NAMESPACE }} -f ${{ env.OG_PATH }}
+
+      - name: Setup subscription yaml
+        uses: mikefarah/yq@master
+        with:
+          cmd: |
+            yq e -i '
+              .spec.channel = "test" |
+              .spec.startingCSV = env(OPERATOR_PKG_NAME) |
+              .spec.name = env(OPERATOR_PKG_NAME) |
+              .spec.source = env(OPERATOR_PKG_NAME)
+              ' ${SUBSCRIPTION_PATH}
+
+      - name: Create subscription
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: create -n ${{ env.NAMESPACE }} -f ${{ env.SUBSCRIPTION_PATH }}
+
+      - name: Deploy Quay
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: create -n ${{ env.NAMESPACE }} -f ${{ env.QUAY_SAMPLE_PATH }}
+
+      - name: Wait for it...
+        run: sleep 10s
+
+      - name: Ensure database rollout
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: rollout -n ${{ env.NAMESPACE }} status deployment skynet-quay-database --timeout=3m
+
+      - name: Ensure redis rollout
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: rollout -n ${{ env.NAMESPACE }} status deployment skynet-quay-redis --timeout=3m
+
+      - name: Ensure config editor rollout
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: rollout -n ${{ env.NAMESPACE }} status deployment skynet-quay-config-editor --timeout=3m
+
+      - name: Ensure Quay rollout
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: rollout -n ${{ env.NAMESPACE }} status deployment skynet-quay-app --timeout=3m
+
+      - name: Ensure mirror rollout
+        uses: actions-hub/kubectl@master
+        env:
+          KUBE_CONFIG: ${{ secrets.KUBE_CONFIG }}
+        with:
+          args: rollout -n ${{ env.NAMESPACE }} status deployment skynet-quay-mirror --timeout=3m
```

Comment:
Can you add a step that pushes/pulls something from Quay? Or some kind of stub for actual e2e tests.

Reply:
I think this would be a nice addition @dmage! I'm not sure exactly which direction to go here however. Since these are e2e tests for the operator itself we'd want the test cases to focus on its own functionality. 
Depending on how good Quay's health check is we might not have to perform any operations against it to know it works.

There's also [`kuttl`](https://kuttl.dev/) that @jonathankingfc brought up, which we can use to build better coverage as we iterate upon our suite.

Reply:
Health checks is a good option, we'll need to check in Quay tests that we can rely on them. Perhaps at this point we can rely on the operator's reportings.

I'd like to see there a step that will execute a shell script/makefile target from the repository. At this point, it can be a simple check that the deployment is available. But it'll give us a check that every engineer can run on their cluster.

Reply:
> Health checks is a good option, we'll need to check them in the Quay tests so that we can rely on them. Perhaps at this point we can simply rely on the operator's reportings.

Agreed. I'll have a look at this and ping you with my findings.

> I'd like to see there a step that will execute a shell script/makefile target from the repository. At this point, it can be a simple check that the deployment is available. But it'll give us a check that every engineer can run on their cluster.

I really like the idea of making commands used in the e2e suite available for anyone to run against their own cluster. From what I gather we can use kuttl for that too - it requires a [kubectl plugin](https://kuttl.dev/docs/#install-kuttl-cli), but I think that isn't so much to ask from a dev wanting to run the e2e suite.
There's a [JIRA](https://issues.redhat.com/browse/PROJQUAY-2594) tracking the work on adding kuttl to the suite.